output/elasticsearch/elasticsearch.go

```diff
@@ -32,6 +32,7 @@ type Indexer struct {
 	indexType         string
 	RateCounter       *ratecounter.RateCounter
 	lastDisplayUpdate time.Time
+	fields            map[string]string
```

Comment:
What indeed. :)

I was just following the established convention that it routes only by strings.  If there is a need to support routing for all JSON datatypes, it could be done with a type switch in each output plugin.  IMO that's overkill unless someone has a compelling reason.